zerver/tests/test_message_send.py

```diff
@@ -2362,6 +2365,86 @@ def test_personal(self) -> None:
             receiver=self.example_user("othello"),
         )
 
+    def test_direct_message_initiator_group_setting(self) -> None:
+        """
+        Tests that direct_message_initiator_group_setting works correctly.
+        """
+        user_profile = self.example_user("hamlet")
+        polonius = self.example_user("polonius")
+        admin = self.example_user("iago")
+        realm = user_profile.realm
+        huddle = [user_profile, admin, polonius]
+        administrators_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.ADMINISTRATORS, realm=realm, is_system_group=True
+        )
+        self.login_user(user_profile)
+        self.send_personal_message(user_profile, polonius)
+        do_change_realm_permission_group_setting(
+            realm,
+            "direct_message_initiator_group",
+            administrators_system_group,
+            acting_user=None,
+        )
+
+        with self.assertRaises(JsonableError):
+            self.send_personal_message(user_profile, self.example_user("cordelia"))
+        with self.assertRaises(JsonableError):
+            self.send_personal_message(user_profile, admin)
+        with self.assertRaises(JsonableError):
+            self.send_huddle_message(user_profile, huddle)
+
+        self.send_personal_message(user_profile, polonius)
+        self.send_personal_message(admin, user_profile)
+        self.send_personal_message(user_profile, admin)
+        self.send_personal_message(user_profile, user_profile)
+
+        self.send_huddle_message(admin, huddle)
+        self.send_huddle_message(user_profile, huddle)
```

Comment:
Move the above block trying to send huddle messages to just above this.

```
        with self.assertRaises(JsonableError):
            self.send_huddle_message(user_profile, huddle)
```

Reply:
The test will not work if I move it above the `assertRaises` block. So, I have added a second huddle (`huddle_2`) to test the error after the messages are sent to `huddle_1`.